posthog/models/cohort.py

```diff
@@ -89,7 +91,68 @@ class Cohort(models.Model):
 
     objects = CohortManager()
 
+    # deprecated
+    groups: models.JSONField = models.JSONField(default=list)
+
+    @property
+    def properties(self) -> PropertyGroup:
+        # convert deprecated groups to properties
+        if self.groups:
+            property_groups = []
+            for group in self.groups:
+                if group.get("properties"):
+                    # Do not try simplifying properties at this stage. We'll let this happen at query time.
+                    property_groups.append(
+                        Filter(data={**group, "is_simplified": True}, team=self.team).property_groups
+                    )
+                elif group.get("action_id"):
+                    property_groups.append(
+                        PropertyGroup(
+                            PropertyOperatorType.AND,
+                            [
+                                Property(
+                                    key=group.get("action_id"),
+                                    type="behavioural",
+                                    value="performed_event_multiple" if group.get("count") else "performed_event",
+                                    event_type="actions",
+                                    time_interval="day",
+                                    time_value=group.get("days"),
+                                    operator=group.get("count_operator"),
+                                    operator_value=group.get("count"),
+                                    negation=group.get("count") == 0 and group.get("count_operator") in ["lte", "eq"],
```

Comment:
Hmm, sure thing, I won't touch negation here then, since there's no way, given new constraints, it will work for any existing old group, as far as I can tell.

Hmm, does raise this question for me, that why do we have these new restrictions on negations?

Reply:
The main reason is that it allows users to do confusing things that they probably don't intend - and the result is a giant cohort (which isn't great). 

For example, if you say "users who never performed an insight analyzed" on our account, you're going to get a result that includes every user who ever visited the marketing page. While this is technically correct, it probably isn't what the user wants.

The user can still get to the same result by saying "users did $pageview  AND who never performed an insight analyzed", but they have to take the one further step showing they understand what they're asking for.